### ISSUE TITLE
Incorrect handling of InterruptedException

### ISSUE THREAD

**Issue by ovkhasch** [AltisourceLabs/ecloudmanager#36]
Polling is now done using Failsafe class. But internally it handles InterruptedException by wrapping it into FailsafeException and throwing it. Our client code doesn't handle FailsafeException. This is a potential problem because thread doesn't get interrupted flag and interrupted state will not be handled by j2ee container correctly. The client code will fail with FailsafeException instead.

Some reading on this:
http://stackoverflow.com/questions/4906799/why-invoke-thread-currentthread-interrupt-when-catch-any-interruptexception
http://stackoverflow.com/questions/8202492/thread-sleep-in-an-ejb

SynchronousPoller used guava RateLimiter which handled InterruptedException internally in a correct way by calling Thread.currentThread().interrupt();

**Comment by vshipb** [comment]
We don't call interrupt() method on threads where Failsafe library is used, so we don't need to worry about this.
http://stackoverflow.com/questions/26856242/interruptedexception-what-causes-it

**Comment by ovkhasch** [comment]
This is not correct. Our threads are managed by the container. It may call interrupt().
